subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/CreateDependencyInfoFileTask.java

```diff
@@ -0,0 +1,97 @@
+package org.shipkit.internal.gradle.java.tasks;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.*;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.shipkit.version.VersionInfo;
+
+import java.io.File;
+
+/**
+ * Creates a file {@link #getOutputFile()} with information about all resolved dependencies of the project.
+ * Dependencies are taken from passed {@link #getConfiguration()}.
+ * Results include:
+ * - resolved dependencies
+ * - files
+ * - gradleApi()
+ * - localGroovy()
+ *
+ * They don't include the sibling projects of the same version. Eg. if project A depends on sibling project B
```

Comment:
When explaining behavior it is worth describing not only what the code does but also why. In this example: why we are not including sibling dependencies?